nemo_skills/code_execution/sandbox.py

```diff
@@ -208,7 +213,12 @@ def execute_code(
     }}
 print(json.dumps(to_return))
 """
-        request = self._prepare_request(TO_EXECUTE, timeout)
+        elif language == 'lean4':
+            TO_EXECUTE = generated_code
```

Comment:
Let's raise an error here is session_id is not None, since we don't support stateful execution for this case

Reply:
I am not sure why session_id could be not None. 
`elif language == 'lean4':
            if session_id is not None:
                raise RuntimeError(f"Stateful execution for {language} is not supported. session_id is not None")
            TO_EXECUTE = generated_code`
do you mean error like this?

Reply:
```
elif language == 'lean4':
            if session_id is not None:
                raise RuntimeError(f"Stateful execution for {language} is not supported. session_id is not None")
            TO_EXECUTE = generated_code
```